src/main/java/com/avispl/dal/communicator/cisco/CiscoCommunicator.java

```diff
@@ -1569,6 +1588,151 @@ private void populateRoomAnalyticsData(Map<String, String> statistics, List<Adva
         }
     }
 
+    /**
+     * Retrieve Peripherals statistics/controls values. In order to do so - we need to perform a control command
+     * unlike data is normally received with status/configuration xml payloads. Status payload only contains connected
+     * devices data, but in oder to retrieve registered devices also - command action is needed.
+     *
+     * @param statistics             map to set statistics data to
+     * @param controllableProperties list of controls to add controllable properties to
+     * @param status                 device status data fetched from {@link #statusPath}
+     * @param configuration          device configuration data fetched from {@link #configurationPath}
+     * @param valuespace             device valuespace data fetched from {@link #valuespacePath}
+     */
+    private void populatePeripheralsData(Map<String, String> statistics, List<AdvancedControllableProperty> controllableProperties,
+                                         CiscoStatus status, CiscoConfiguration configuration, String valuespace) throws Exception {
+        // This is a unique case of retrieval statistics data, so it is not exposed to a separate method
+        Command command = new Command();
+        PeripheralsCommand peripheralsCommand = new PeripheralsCommand();
+        peripheralsCommand.setList(new ListCommand());
+        command.setPeripheralsCommand(peripheralsCommand);
+        Command peripheralsListCommandResponse = doPost("putxml", command, Command.class);
+
+        Peripherals peripheralsStatus = status.getPeriherals();
+        List<ConnectedDevice> connectedDevices = new ArrayList<>();
+        if (peripheralsStatus != null) {
+            connectedDevices.addAll(Arrays.asList(peripheralsStatus.getConnectedDevice()));
+        }
+
+        if (peripheralsListCommandResponse != null) {
+            PeripheralsListResult peripheralsListResult = peripheralsListCommandResponse.getPeripheralsListResult();
+            if (peripheralsListResult != null && "OK".equals(peripheralsListResult.getStatus())) {
```

Comment:
is it guaranteed that keyword "OK" is always returned in upper-case format? why not `equalsIgnoreCase`